extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/videos/GoogleVideosImporter.java

```diff
@@ -94,54 +99,99 @@ public ImportResult importItem(
       return ImportResult.OK;
     }
 
+    PhotosLibrarySettings settings =
+        PhotosLibrarySettings.newBuilder()
+            .setCredentialsProvider(
+                FixedCredentialsProvider.create(
+                    UserCredentials.newBuilder()
+                        .setClientId(appCredentials.getKey())
+                        .setClientSecret(appCredentials.getSecret())
+                        .setAccessToken(new AccessToken(authData.getAccessToken(), null))
+                        .setRefreshToken(authData.getRefreshToken())
+                        .build()))
+            .build();
+
     //     Uploads videos
     if (data.getVideos() != null && data.getVideos().size() > 0) {
       for (VideoObject video : data.getVideos()) {
         executor.executeAndSwallowIOExceptions(
-            video.getDataId(), video.getName(), () -> importSingleVideo(authData, video));
+            video.getDataId(), video.getName(), () -> importSingleVideo(video, settings));
       }
     }
     return ImportResult.OK;
   }
 
-  String importSingleVideo(TokensAndUrlAuthData authData, VideoObject inputVideo) throws Exception {
-    // download video and create input stream
-    InputStream inputStream;
+  String importSingleVideo(VideoObject inputVideo, PhotosLibrarySettings settings)
+      throws Exception {
     if (inputVideo.getContentUrl() == null) {
       monitor.info(() -> "Content Url is empty. Make sure that you provide a valid content Url.");
       return null;
     }
 
-    inputStream = this.videoStreamProvider.get(inputVideo.getContentUrl().toString());
-
     String filename;
     if (Strings.isNullOrEmpty(inputVideo.getName())) {
       filename = "untitled";
     } else {
       filename = COPY_PREFIX + inputVideo.getName();
     }
+    File tmp = File.createTempFile(filename, ".mp4");
```

Comment:
Hmm, this just uses a local file, I don't think we make any guarantees/assumptions of how much disk there is on the worker, it seems like you should temp store here right?

Reply:
Yes, that certainly seemed like it would be preferable but UploadMediaItemRequest only supports reading the data from a RandomAccessFile which limits our options. https://google.github.io/java-photoslibrary/1.4.0/com/google/photos/library/v1/upload/UploadMediaItemRequest.Builder.html I don't know if this might be suitable for a later improvement to the Google Photos Library.

Reply:
It seems like the thing we might have to do then is add a method to our JobStore that produces a RandomAccessFile then, that way it can get stored in the persistent storage provided by the cloud extension.

Reply:
Ok, I've created a default implementation of getTempFileFromInputStream on TemporaryPerJobDataStore.